torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -116,6 +116,63 @@ def _call_function_and_unflatten_output(tx, fn, args, kwargs, ret_vt, ret_treesp
     )
 
 
+# NOTE:
+# Mutation detection: We detect mutations by checking the example_value's _version
+# This approach has a corner case that if torch.compile is wrapped
+# by an inference_mode(True) context manager, the tensor version will not be upated.
+# So we override the inference_mode to False then run the extracted graph module.
+#
+# Aliasing detection: We detect mutations by checking the example_value's _typed_storage():
+# if output fake_tensor is sharing the same storage with one of the inputs, we consider it aliasing
+def _detect_input_mutations_and_aliasing(tx, graph_module, proxy_args):
+    from torch.multiprocessing.reductions import StorageWeakRef
+
+    orig_example_values = [arg.node.meta["example_value"] for arg in proxy_args]
+    with torch.inference_mode(False), tx.fake_mode, enable_python_dispatcher():
+        # We clone the tensors here because:
+        # 1. if torch.inference_mode(True), the orig_example_values won't have
+        # the _version attribute.
+        # 2. we don't want to mutate the original example values' _version.
+        example_values = [t.clone() for t in orig_example_values]
+        _before_versions = [ex._version for ex in example_values]
+        res = graph_module(*example_values)
```

Comment:
Er, what happens if there are side effects in `graph_module`? E.g. torch.set_grad_enabled("False") may appear there.

If you can make an argument that this was a problem previously, we can ship this PR, otherwise we probably want to think more about how to get this case to work.

Reply:
A naive solution is that we check whether there are state manipulate ops inside the graph module before executing it and raise unimplemented if there is. 

But it seems a bit difficult to come up with a complete list. Maybe we could start with banning all operators that are not higher order op nor aten ops nor built-in ops, and gradually added more in the allowed list. We could do some state backup/restoring for those in allowed list.